coins/monero/src/ringct/mlsag/mod.rs

```diff
@@ -0,0 +1,114 @@
+#![allow(non_snake_case)]
+
+use std::io;
+use std::io::{Read, Write};
+
+use curve25519_dalek::scalar::Scalar;
+use curve25519_dalek::edwards::EdwardsPoint;
+use curve25519_dalek::traits::Identity;
+
+use crate::{hash_to_scalar, serialize::*};
+use crate::ringct::hash_to_point;
+
+/// MgSig part of MLSAG, as used in Monero.
```

Comment:
This isn't the MgSig part. This IS the MLSAG, per https://eprint.iacr.org/2015/1098.pdf.

Reply:
This is how its called in monero-rs

+ https://github.com/monero-project/monero/blob/0a1eaf26f9dd6b762c2582ee12603b2a4671c735/src/ringct/rctTypes.h#L168

so MgSig is just that + II ?

Reply:
Monero notates it MgSig. That doesn't change it is the MLSAG from Monero.

The II is the list of key images. Annoyingly, the IACR paper merged it (part of the statement) with the proof. I'd delineate the MLSAG proof as solely the c scalar and ss matrix, which is what's here, unless I'm missing something.